libenv/unix_iface.c

```diff
@@ -197,18 +196,16 @@ static void GetMacAddress(EvalContext *ctx, ARG_UNUSED int fd, struct ifreq *ifr
     {
         if ( strcmp(ifa->ifa_name, ifp->ifr_name) == 0)
         {
-            if (ifa->ifa_addr->sa_family == AF_LINK)
+            if (ifa->ifa_addr->sa_family == AF_PACKET)
```

Comment:
I'm not 100% on this change but think that `man getifaddrs` bolsters the change from AF_LINK to AF_PACKAGE at least for linux.

```
NOTES
       The  addresses returned on Linux will usually be the IPv4 and IPv6 addresses assigned to the interface, but also one AF_PACKET address per interface
       containing lower-level details about the interface and its physical layer.  In this case, the ifa_data field may  contain  a  pointer  to  a  struct
       rtnl_link_stats,  defined  in  <linux/if_link.h> (in Linux 2.4 and earlier, struct net_device_stats, defined in <linux/netdevice.h>), which contains
       various interface attributes and statistics.
```

Reply:
Figured this out by looking at linux/freebsd man pages and results. On android there are ifa items which don't have ifa_addr pointers, so just have to skip where ifa_addr == NULL and keep the code the same otherwise. In progress.

Reply:
should be sorted now. Need to treat things different for BSDs and Android/Termux which doesn't have privs to do ioctl. Depend on AF_LINK of AF_PACKET and if neither (shouldn't happen?) then error out during compile.